pandas/tseries/offsets.py

```diff
@@ -2748,14 +2751,20 @@ def nanos(self):
 
     def apply(self, other):
         # Timestamp can handle tz and nano sec, thus no need to use apply_wraps
-        if isinstance(other, (datetime, np.datetime64, date)):
+        if isinstance(other, Timestamp):
+            result = other.__add__(self)
```

Comment:
added a comment. The + causes it to segfault. I *think* its because of a fast-path in the interpreter, or maybe a cython bug. If there is an exception during the the operation. Then we compound this by trying the reversed ops in ``.apply`` and the cycle repeats. But I think its too hard (maybe impossible) to change this.